.agents/architecture/dual-path-strategy.md

```diff
@@ -0,0 +1,253 @@
+# Dual-Path GitHub Operations Strategy
+
+**Date**: 2025-12-23
+**Principle**: Por qué no los dos?
+
+---
+
+## Platform Capabilities Matrix
+
+| Platform | Skills Support | MCP Support | Best Implementation |
+|----------|---------------|-------------|---------------------|
+| **Claude Code (CLI)** | ✅ Yes | ✅ Yes | GitHub MCP Skill (5-20ms) |
+| **VS Code Agents** | ✅ Yes | ✅ Yes | GitHub MCP Skill (5-20ms) |
+| **Copilot CLI** | ❌ No | ⚠️ Limited | gh CLI wrappers (50-80ms) |
+
+---
+
+## Strategy: Both Implementations
+
+### Path A: GitHub MCP Skill (Primary)
+
+**For**: Claude Code, VS Code Agents
+
+```
+.claude/skills/github/
+├── SKILL.md              # allowed-tools: mcp__github__*
+├── README.md             # Usage guide
+└── docs/
+    ├── pr-operations.md
+    ├── issue-operations.md
+    └── reactions.md
+```
+
+**Performance**: 5-20ms overhead + API time
+**Maintenance**: Low (official GitHub MCP)
+**Coverage**: ~90% (validate reactions/threads)
+
+### Path B: Native Tool Wrappers (Fallback)
+
+**For**: Copilot CLI, bash environments
+
+```
+.claude/skills/github/scripts/
+├── gh-native/
+│   ├── get-pr-context.sh
+│   ├── set-issue-labels.sh
+│   ├── post-issue-comment.sh
+│   └── add-comment-reaction.sh
+└── README.md
+```
+
+**Performance**: 50-80ms overhead + API time
+**Maintenance**: Medium (bash scripts)
+**Coverage**: 100% (gh CLI + graphql)
+
+---
+
+## Routing Logic
+
+### Automatic Detection
+
+```markdown
+## Usage
+
+The github skill automatically selects the best implementation:
```

Comment:
@rjmurillo-bot We'd have to make sure the agent is passing in information about the environment in ordrer to make this work. For example, they may not have the MCP configured, so we'd just fall back to the gh wrappers. This is another scenario that needs UltraThink